File: 88.MergeSortedArray/88.MergeSortedArray.js
var merge = function (nums1, m, nums2, n) {
  let i = m - 1;
  let j = n - 1;
  let k = m + n - 1;

  while (k > 0) {
    if (nums1[i] < nums2[j]) {
      nums1[k] = nums2.pop();
      j--;
    } else {
      nums1[k] = nums1[i];
      i--;
    }
    k--;
  }

  for (let i = 0; i < nums2.length; i++) {
    nums1[i] = nums2[i];
  }
};

console.log(merge([1, 2, 3, 0, 0, 0], 3, [2, 5, 6], 3));

File: 88.MergeSortedArray/88.MergeSortedArray.py
def merge(nums1, m, nums2, n):
  l = len(nums1)-1

  while m > 0 and n > 0:
    if nums2[n-1] > nums1[m-1]:
      nums1[l] = nums2[n-1]
      n -=1
    else:
      nums1[l] = nums1[m-1]
      m -= 1
    l -= 1
  
  while n > 0:
    nums1[l] = nums2[n-1]
    n -= 1
    l -= 1
  print(nums1)
  

merge([0], 0, [1], 1)